source/opt/set_spec_constant_default_value_pass.cpp

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "set_spec_constant_default_value_pass.h"
+
+#include <cstring>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+
+#include "def_use_manager.h"
+#include "spirv-tools/libspirv.h"
```

Comment:
Nitpicky:  Order of includes:  Other libraries headers come before your own libraries.
The optimization library is separate from libSPIRV-Tools.a.

(I haven't been checking this bit very closely at all lately.)